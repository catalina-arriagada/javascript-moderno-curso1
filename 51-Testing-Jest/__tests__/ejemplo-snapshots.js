const cliente = {
    nombre: 'jp', //npm t -- -u //en la terminal, actualiza el snapshop completo si quiero modificar el objeto
    balance: 500,
    tipo:'premium'
};

describe('testing al cliente', () => {
    test('es jp', () => {
        expect(cliente.nombre).toBe('jp');
    });

    //probar objeto completo: convertir a json string pero existe en jest algo llamado los snapshots, que son datos que se crean en carpeta de strings y sobre ella podemos comparar si es el mismo dato u otro
    test('es igual al objeto completo', () => {
        expect(cliente).toMatchSnapshot();
    });
})