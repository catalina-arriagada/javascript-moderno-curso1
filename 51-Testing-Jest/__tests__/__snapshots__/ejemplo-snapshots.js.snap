// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing al cliente es igual al objeto completo 1`] = `
{
  "balance": 500,
  "nombre": "jp",
  "tipo": "premium",
}
`;
